pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java

```diff
@@ -245,6 +252,12 @@ public boolean index(GenericRow row) {
     minTimeVal = Math.min(minTimeVal, timeValue);
     maxTimeVal = Math.max(maxTimeVal, timeValue);
 
+    try {
+      long msgEventTimeLagInSeconds = System.currentTimeMillis() / 1000 - (long) toSecondsTimeConverter.convert(timeValueObj);
+      serverMetrics.addTimeredTableValue(tableAndStreamName, ServerTimer.CURRENT_MSG_EVENT_TIMESTAMP_LAG, msgEventTimeLagInSeconds, TimeUnit.SECONDS);
```

Comment:
In incoming time granularity is days, we will get a saw-tooth graph, am I right?
Alternative would be to compare according to the time granularity, so that we get a flat straight line. Of course, we need to add some fuzziness to account for the events that come in next day, and allow them to come in for _some_ time.